src/SolutionAcceleration.f90

```diff
@@ -0,0 +1,395 @@
+TYPE,EXTENDS(RelaxedPicardType) :: ModifiedPicardType
+  !> Temporary vector to hold previous solution
+  CLASS(VectorType),ALLOCATABLE :: tmpvec
+!
+!List of Type Bound Procedures
+  CONTAINS
+    !> @copybrief ModifiedPicardType::clear_ModifiedPicardType
+    !> @copydetails ModifiedPicardType::clear_ModifiedPicardType
+    PROCEDURE,PASS :: clear => clear_ModifiedPicardType
+    !> @copybrief ModifiedPicardType::setInitial_ModifiedPicardType
+    !> @copydetails ModifiedPicardType::setInitial_ModifiedPicardType
+    PROCEDURE,PASS :: setInitial => setInitial_ModifiedPicardType
+    !> @copybrief ModifiedPicardType::step_ModifiedPicardType
+    !> @copydetails ModifiedPicardType::step_ModifiedPicardType
+    PROCEDURE,PASS :: step => step_ModifiedPicardType
+ENDTYPE ModifiedPicardType
+
+!> Name of module
+CHARACTER(LEN=*),PARAMETER :: modName='SolutionAccelerationModules'
+!
+!===============================================================================
+CONTAINS
+!
+!-------------------------------------------------------------------------------
+!> @brief Initializes the Solution Acceleration base class
+!> @param solver The Solution Acceleration solver to act on
+!> @param Params The options parameter list
+!> @param ce The computing environment to use for the calculation
+!>
+SUBROUTINE init_base_SolutionAccelerationType(solver,ce,Params)
+  CLASS(SolutionAccelerationType),INTENT(INOUT) :: solver
+  TYPE(FutilityComputingEnvironment),TARGET,INTENT(IN) :: ce
+  TYPE(ParamType),INTENT(IN) :: Params
+
+  CHARACTER(LEN=*),PARAMETER :: myName='init_base_SolutionAccelerationType'
+
+  REQUIRE(Params%has('SolutionAccelerationType->n'))
+  REQUIRE(.NOT.solver%isInit)
+
+  !Pull Data from Parameter List
+  CALL Params%get('SolutionAccelerationType->n',solver%n)
+  IF(Params%has('SolutionAccelerationType->start')) &
+      CALL Params%get('SolutionAccelerationType->start',solver%start)
+
+  IF(solver%n < 1) CALL ce%exceptHandler%raiseError('Incorrect input to '//modName// &
+      '::'//myName//' - Number of unkowns (n) must be greater than 0!')
+
+  IF(solver%start < 1) CALL ce%exceptHandler%raiseError('Incorrect input to '//modName// &
+      '::'//myName//' - Starting point must be greater than 1!')
+
+  solver%s=0
+  solver%ce => ce
+ENDSUBROUTINE init_base_SolutionAccelerationType
+!
+!-------------------------------------------------------------------------------
+!> @brief Clears the solution acceleration base class
+!> @param solver The solver to act on
+!>
+SUBROUTINE clear_base_SolutionAccelerationType(solver)
+    CLASS(SolutionAccelerationType),INTENT(INOUT) :: solver
+
+    INTEGER(SIK) :: i
+
+    IF(solver%isInit) THEN
+      IF(ALLOCATED(solver%x)) THEN
+        DO i=1,SIZE(solver%x)
+          CALL solver%x(i)%clear()
+        ENDDO
+        DEALLOCATE(solver%x)
+      ENDIF
+      solver%s=0
+      solver%n=-1
+      solver%start=1
+    ENDIF
+  ENDSUBROUTINE clear_base_SolutionAccelerationType
```

Comment:
Need to set `isInit` to false

Reply:
I thought it should be part of the extending class to set isInit to false.  This is why I didn't duplicate it as the main clear routine.  My view is isInit shouldn't be set to false until everything is cleared.

Reply:
Hmm... my view is that `isInit` should only be true if the object is fully initialized... and therefore can be set to false if any part of the object has been cleared...

Reply:
Yeah, that's fair enough.

Reply:
Fixed